cmake/BipedalLocomotionFrameworkDependencies.cmake

```diff
@@ -110,7 +113,7 @@ framework_dependent_option(FRAMEWORK_COMPILE_Contact
 
 framework_dependent_option(FRAMEWORK_COMPILE_Planners
   "Compile Planners libraries?" ON
-  "FRAMEWORK_USE_Qhull;FRAMEWORK_USE_casadi;FRAMEWORK_USE_manif;FRAMEWORK_COMPILE_Math" OFF)
+  "FRAMEWORK_USE_Qhull;FRAMEWORK_USE_casadi;FRAMEWORK_USE_manif;FRAMEWORK_COMPILE_Math;FRAMEWORK_USE_UnicyclePlanner" OFF)
```

Comment:
I would avoid setting this dependency for the whole Planners part. Would it make sense to create a separate library, similarly to what we do for the conversions and the parameters handlers?

Reply:
This is what I did initially, then you suggested to include them in the Planners and I changed :) Anyway, this is only temporary until when the unicycle classes will get vendored in blf, right?

Reply:
Sorry, I meant in the ``Planners`` folder but in a different library. Really sorry for the confusion

Reply:
If we proceed this way, wouldn't the `Planners` component still need the `UnicyclePlanner` dependency? Or you had something else in mind? The only way to cleanly separate things is a brand new component as it was in the very beginning.

Reply:
I was thinking of something similar to https://github.com/dic-iit/bipedal-locomotion-framework/blob/bd401fee1cae092c089536ef93c7288db1cb0a40/src/Conversions/CMakeLists.txt#L14-L32 for example, where, depending on the available dependencies, we enable the compilation of specific libraries. The drawback is that the target cannot be the same, i.e. we should have a target like ``UnicyclePlanner`` that is compiled from the ``Planners`` folder.

Reply:
Done in a0b66a5 and 777ca69

Reply:
Is there a specific reason for keeping this asymmetry?

Reply:
Yes, differently from the library names that could be different for Planner and Unicycle, the Python module has to be compiled in the same library. This is because the library creates a module `bipedal_locomotion_framework.bindings.planners`. If we want, we can take a macro approach, and I mean adding a new compiler definition that depends on whether the Unicycle target was compiled, and alter the sources list of the pybind11 target. Otherwise, thing could remain as they are right now and the planner bindings get compiled only if also the unicycle is enabled.

Reply:
Ok clear! We can keep this in mind for the future